woocommerce/woocommerce
# Delayed Account Creation: Support option to send password setup link to customer via email

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Respects the `woocommerce_registration_generate_password` setting by hiding the password field and sending the link via email.

Note, this branch is based on https://github.com/woocommerce/woocommerce/pull/50934 so review https://github.com/woocommerce/woocommerce/pull/50934 first.

Closes #51193



### How to test the changes in this Pull Request:

**This block is behind an experimental flag. If you are testing on a production version of WooCommerce this block should not be available (yet).** You can however test the reset password flow:

1. Go to the my account page while logged out and reset password
2. Click the link you get via email and set a new password
3. Confirm you are logged in

These instructions are for testing the development build:

1. Appearance > Templates > WooCommerce > Order Confirmation
4. Insert the Account Creation block below the order summary (this is were it will be auto inserted in the future). Save.
5. Confirm the editor shows a non-interactive form and an editable list.
6. Go to WooCommerce > Settings > Accounts and privacy > Check the "Send password setup link (recommended)" option and save.
7. Log-out and checkout as a guest.
8. On the order confirmation page, try the "create account" button. There should be no password field visible.
9. Verify account was created successfully and you get a confirmation.
10. Verify going to account page only shows your latest order.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Delayed account creation: Support option to send password setup link to customer via email.

</details>
